src/cipher.rs

```diff
@@ -517,10 +519,11 @@ pub struct rustls_client_cert_verifier {
     _private: [u8; 0],
 }
 
-impl CastConstPtr for rustls_client_cert_verifier {
+impl CastPtr for rustls_client_cert_verifier {
     type RustType = AllowAnyAuthenticatedClient;
 }
 
+impl BoxCastPtr for rustls_client_cert_verifier {}
 impl ArcCastPtr for rustls_client_cert_verifier {}
```

Comment:
I'm a little bit skeptical these cahnges to the `*Ptr` traits for the verifiers are correct. Earmarked to dig into the associated macros for closer study.

In particular: worried the change from Arc -> Box on the constructor side will interact poorly with multiple handles to a verifier being used to add CRLs, then being turned back to Arc to give to Webpki :thinking: